### ISSUE TITLE
mutate_if() claim to ignore grouping variables, but it does not

### ISSUE THREAD

**Issue by courtiol** [tidyverse/dplyr#4572]
Hi all,
for some reason, mutate_if() says it ignores grouping variables but the following reprex clearly shows that it does not ignore it. If that is meant to say that the mutation don't affect the grouping variable as such, then it is curious that other functions do not show such message. So I think this may be a glitch.

``` r
library(tidyverse)
#> ── Attaching packages ─────────────────────────────────────────────────────────────────────────────── tidyverse 1.2.1 ──
#> ✔ ggplot2 3.2.1          ✔ purrr   0.3.2     
#> ✔ tibble  2.1.3          ✔ dplyr   0.8.3.9000
#> ✔ tidyr   1.0.0          ✔ stringr 1.4.0     
#> ✔ readr   1.3.1          ✔ forcats 0.4.0
#> ── Conflicts ────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
#> ✖ dplyr::filter() masks stats::filter()
#> ✖ dplyr::lag()    masks stats::lag()
iris %>%
  group_by(Species) %>%
  mutate_if(is.numeric, mean) %>%
  slice(1:2)
#> `mutate_if()` ignored the following grouping variables:
#> Column `Species`
#> # A tibble: 6 x 5
#> # Groups:   Species [3]
#>   Sepal.Length Sepal.Width Petal.Length Petal.Width Species   
#>          <dbl>       <dbl>        <dbl>       <dbl> <fct>     
#> 1         5.01        3.43         1.46       0.246 setosa    
#> 2         5.01        3.43         1.46       0.246 setosa    
#> 3         5.94        2.77         4.26       1.33  versicolor
#> 4         5.94        2.77         4.26       1.33  versicolor
#> 5         6.59        2.97         5.55       2.03  virginica 
#> 6         6.59        2.97         5.55       2.03  virginica
```

<sup>Created on 2019-09-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>
++
Alex

**Comment by courtiol** [comment]
Perhaps this behaviour is consistent with the doc for `?dplyr::scoped`:

* _Summarising operations (summarise_all() and summarise_if()) ignore grouping variables silently because it is obvious that operations are not applied on grouping variables._
* _On the other hand it isn't as obvious in the case of mutating operations (mutate_all(), mutate_if(), transmute_all(), and transmute_if()). For this reason, they issue a message indicating which grouping variables are ignored._

But then the message should perhaps be ```'mutate_if()' may have ignored the following grouping variables```. No?

**Comment by hadley** [comment]
Yeah, I think we just need to tweak the message here.

**Comment by hadley** [comment]
Minimal reprex:

``` r
library(dplyr, warn.conflicts = FALSE)
df <- tibble(x = 1, y = 2)

df %>% 
  group_by(x) %>% 
  mutate_if(is.numeric, mean)
#> `mutate_if()` ignored the following grouping variables:
#> Column `x`
#> # A tibble: 1 x 2
#> # Groups:   x [1]
#>       x     y
#>   <dbl> <dbl>
#> 1     1     2

df %>%
  group_by(x) %>%
  mutate(across(is.numeric, mean))
#> Error: Column `x` can't be modified because it's a grouping variable
```

<sup>Created on 2020-01-05 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

@romainfrancois should `across()` ignore grouping variables by default?

**Comment by hadley** [comment]
We've decided that `across()` will always ignore grouping variables — this makes it is more consistent and easier to reason about, and I'm pretty is what you almost always want. Since we're extremely likely retire the scoped helped in favour of `across()`, this behaviour will remain in `mutate_if()`, but will be fixed in our preferred alternative.

**Comment by hadley** [comment]
Fixed in #4700
